wpilib/src/dio.rs

```diff
@@ -154,6 +123,55 @@ impl OutputPin for DigitalOutput {
     }
 }
 
+/// Digital PWM output.
+///
+/// PWM output will be disabled when this is dropped.
+///
+/// Use [`DigitalOutput::enable_pwm`] to get an instance of this.
+///
+/// [`DigitalOutput::enable_pwm`]: struct.DigitalOutput.html#method.enable_pwm
+pub struct DigitalPwm {
+    pin: DigitalOutput,
+    pwm: DigitalPwmHandle,
```

Comment:
I'm vaguely concerned about the drop order. [The reference](https://doc.rust-lang.org/stable/reference/destructors.html) says members are dropped in declaration order, meaning the `DigitalOut` goes before it's `DigitalPwmHandle`. Not sure if that's a problem. WPILibC's destructor _does_ call `DisablePWM()` before `HAL_FreeDIOPort`:
```
wpilibc/src/main/native/cpp/DigitalOutput.cpp
47:DigitalOutput::~DigitalOutput() {
48-  if (StatusIsFatal()) return;
49-  // Disable the PWM in case it was running.
50-  DisablePWM();
51-
52-  HAL_FreeDIOPort(m_handle);
53-}
```
Given that, I think it's best to reverse the field order here, and add a comment saying why and noting that it must stay the same.

Reply:
Good catch, thanks.